tools/cli/domainCommands.go

```diff
@@ -901,14 +913,68 @@ func (d *domainCLIImpl) migrationDynamicConfigCheck(c *cli.Context) DomainMigrat
 
 			if currResp.Value != newResp.Value {
 				check = false
-				mismatchedConfigs = append(mismatchedConfigs, MismatchedDynamicConfig{
-					CurrResp: currResp,
-					NewResp:  newResp,
+				mismatchedConfig = MismatchedDynamicConfig{
+					Key: configKey,
+					CurrValues: []*types.DynamicConfigValue{
+						toDynamicConfigValue(currResp.Value, map[dc.Filter]interface{}{
+							dynamicconfig.DomainID: currentDomainID,
+						}),
+					},
+					NewValues: []*types.DynamicConfigValue{
+						toDynamicConfigValue(currResp.Value, map[dc.Filter]interface{}{
+							dynamicconfig.DomainID: destinationDomainID,
+						}),
+					},
+				}
+			}
+
+		} else if containsFilter(configKey, dc.DomainName.String()) && containsFilter(configKey, dc.TaskListName.String()) {
+			taskLists := c.StringSlice(FlagTaskList)
+			for _, taskList := range taskLists {
+
+				currRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configKey.String(), []dynamicconfig.FilterOption{
+					dynamicconfig.DomainFilter(currDomain),
+					dynamicconfig.TaskListFilter(taskList),
+				})
+
+				newRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configKey.String(), []dynamicconfig.FilterOption{
+					dynamicconfig.DomainFilter(newDomain),
+					dynamicconfig.TaskListFilter(taskList),
 				})
+
+				currResp, err := d.frontendAdminClient.GetDynamicConfig(ctx, currRequest)
+				if err != nil {
+					currResp = &types.GetDynamicConfigResponse{}
+				}
+				newResp, err := d.destinationAdminClient.GetDynamicConfig(ctx, newRequest)
+				if err != nil {
+					newResp = &types.GetDynamicConfigResponse{}
+				}
+				if currResp.Value != newResp.Value {
+					check = false
+
+					// create a new MismatchedDynamicConfig entity for each mismatch
+					mismatchedConfig := MismatchedDynamicConfig{
```

Comment:
append mismatchedValues here

```suggestion
					mismatchedCurValues = append(mismatchedCurValues, toDynamicConfigValue(/*fill in*/))
					mismatchedNewValues = append(mismatchedNewValues, toDynamicConfigValue(/*fill in*/))
```